tests/core/server/test_rate_limits.py

```diff
@@ -217,3 +217,47 @@ async def test_percentage_limits(self):
             if not response:
                 saw_disconnect = True
         assert saw_disconnect
+
+    @pytest.mark.asyncio
+    async def test_too_many_outgoing_messages(self):
+        # Too many messages
+        r = RateLimiter(incoming=False)
+        new_peers_message = make_msg(ProtocolMessageTypes.respond_peers, bytes([1]))
+
+        passed = 0
+        blocked = 0
+        for i in range(NON_TX_FREQ):
+            if r.process_msg_and_check(new_peers_message):
+                passed += 1
+            else:
+                blocked += 1
+
+        assert passed == 10
+        assert blocked == NON_TX_FREQ - passed
+
+        # ensure that *another* message type is not blocked because of this
+
+        new_signatures_message = make_msg(ProtocolMessageTypes.respond_signatures, bytes([1]))
+        assert r.process_msg_and_check(new_signatures_message)
+
+    @pytest.mark.asyncio
+    async def test_too_many_incoming_messages(self):
+        # Too many messages
+        r = RateLimiter(incoming=True)
+        new_peers_message = make_msg(ProtocolMessageTypes.respond_peers, bytes([1]))
+
+        passed = 0
+        blocked = 0
+        for i in range(NON_TX_FREQ):
+            if r.process_msg_and_check(new_peers_message):
+                passed += 1
+            else:
+                blocked += 1
+
+        assert passed == 10
```

Comment:
Why is passed == 10 here?

Reply:
ah, the rate limit for this type of message is 10 (per the unit of time specified in the `RateLimit` constructor)